src/Writer/PhpArray.php

```diff
@@ -139,4 +158,54 @@ protected function processIndented(array $config, array $arraySyntax, &$indentLe
 
         return $arrayString;
     }
+
+    /**
+     * Process a string configuration value
+     *
+     * @param string $value
+     * @return string
+     */
+    protected function processStringValue($value)
+    {
+        if ($this->useClassNameScalars && $this->checkStringIsFqn($value)) {
+            return $value . '::class';
+        }
+
+        return var_export($value, true);
+    }
+
+    /**
+     * Process a string configuration key
+     *
+     * @param string $key
+     * @return string
+     */
+    protected function processStringKey($key)
+    {
+        if ($this->useClassNameScalars && $this->checkStringIsFqn($key)) {
+            return $key . '::class';
+        }
+
+        return "'" . addslashes($key) . "'";
+    }
+
+    /**
+     * Check whether a string represents a resolvable FQCN
+     *
+     * @param string $string
+     * @return bool
+     */
+    protected function checkStringIsFqn($string)
```

Comment:
Funny. I had same thoughts whether to use regex check or not.

My conclusion was:
* that `checkStringIsFqn` is used internally and does not need to expose full scale functionality of detecting a fqn. Only as much as needed internal.
* To ensure we do not have a collision with relative name resolution we simply need to prepend a backslash.
* Adding a regex check might blow it up unnecessary

I went through this now again after you mentioned and my Idea about regex might be absolute wrong. If you use autoloading and check against different strings you always trigger autoloading, which might be worser. 

Is  this your main concern?